R/g.part5.definedays.R

```diff
@@ -74,26 +75,36 @@ g.part5.definedays = function(nightsi, wi, indjump, nightsi_bu,
         if (qwindow[1] != 0) qwindow = c(0, qwindow)
         if (qwindow[length(qwindow)] != 24) qwindow = c(qwindow, 24)
       }
+      # define segments names
       breaks = qwindow2timestamp(qwindow, epochSize)
+      startOfSegments = breaks[-length(breaks)]
+      endOfSegments = subtractEpochFromTimeName(breaks[-1], epochSize)
+      segments_timing = paste(startOfSegments, endOfSegments, sep = "-")
+      if (length(startOfSegments) > 1) {
+        # then, add full MM window
+        segments_timing = c(paste(firstepoch, lastepoch, sep = "-"), segments_timing)
+      }
       if (24 %in% qwindow) {
         # 24:00:00: probably does not exist, replace by last timestamp in a day
-        # here, we consider N epochs per day plus 1 hour just in case we are deriving this in 
+        # here, we consider N epochs per day plus 1 hour just in case we are deriving this in
         # a 25-hour daylight saving time day
         NepochPerDayPlusOneHr = ((25*3600) / epochSize)
         latest_time_in_day = max(format(ts$time[1:pmin(Nts, NepochPerDayPlusOneHr)], format = "%H:%M:%S"))
         breaks = gsub(pattern = "24:00:00", replacement = latest_time_in_day, x = breaks)
       }
-      breaks_i = c()
+      breaks_i = breaks_available = c()
```

Comment:
Please replace `c()` by `rep(NA, length(breaks))` to declare expected object size before loop. Maybe not essential here for speed, but it is a good habit computing efficiency wise to declare object size before loop instead of letting them grow inside a loop.